src/main/webapp/static/js/course_service.js

```diff
@@ -0,0 +1,85 @@
+(function() {
+	'use strict';
+	angular.module('myApp')
+		.factory('CourseService', CourseService);
+		
+	CourseService.$inject = ['$http', '$log'];
+	
+	function CourseService($http, $log) {
+		var REST_SERVICE_URI = 'http://localhost:8080/rest/courses'; 
+		
+		var service = {
+	        fetchAllCourses: fetchAllCourses,
+	        createCourse: createCourse,
+	        updateCourse: updateCourse,
+	        deleteCourse: deleteCourse
+    	};
+
+		return service;
+		
+		function fetchAllCourses() {
+	 
+	      return $http.get(REST_SERVICE_URI)
+	            .then(
+	            function (response) {
+				$log.debug("service data :: "+ JSON.stringify(response));
+	    
+					return response.data;
+	
+	            },
+	            function(errResponse){
+	                $log.error('Error while fetching Courses');
+					//optional error handling in UI
+	            }
+	        );
+
+	       
+	    }
+	 
+	 	function createCourse(user) {
+	        var deferred = $q.defer();
```

Comment:
$q isn't being injected. I don't think this will work.  Regardless, you don't need to use the $q library.